### PR TITLE
Add the checkpoint for the pretrained model

### PR DESCRIPTION
**Pull request by wawltor**
### PR types
 New features

### PR changes
APIs
### Describe
Add the checkpoint for the pretrained model, those state will load from the checkpoint 
1)   optimizer state 
2)  learning_rate message 
3)  random state 
4)  global_step & checkpoint_time 
5)  training arguments

### CODE REVIEW

File: examples/language_model/bert/run_pretrain.py [hunk 1]
```diff
@@ -414,14 +421,13 @@ def do_train(args):
                                                   "model_%d" % global_step)
                         if not os.path.exists(output_dir):
                             os.makedirs(output_dir)
-                        # need better way to get inner model of DataParallel
                         model_to_save = model._layers if isinstance(
                             model, paddle.DataParallel) else model
                         model_to_save.save_pretrained(output_dir)
+                        checkpoint.save_checkpoint(output_dir, global_step,
```
Comment (ZeyuChen) [inline examples/language_model/bert/run_pretrain.py:427]: checkpoint.save_checkpoint的动作语义就不太自然
应该是model.save_checkpoint比较合理，理想的语义是代码即注释。

File: paddlenlp/utils/checkpoint.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
+# coding:utf-8
```
Comment (ZeyuChen) [inline paddlenlp/utils/checkpoint.py:1]: 总感觉单独的checkpoint设计，脱离Trainer的考虑不是特别自然

File: examples/language_model/bert/run_pretrain.py (hunk 1 above)
Comment (ZeyuChen) [reply examples/language_model/bert/run_pretrain.py:427]: 感觉这里的搭配用法非常奇怪，既要model_to_save, 又要checkpoint.save

**Review by ZeyuChen — CHANGES_REQUESTED** [review]
Need further discussion, seems not so natural.

File: examples/language_model/bert/run_pretrain.py (hunk 1 above)
Comment (wawltor) [reply examples/language_model/bert/run_pretrain.py:427]: 这里的checkpoint主要是checkpoint optimizer和learning_rate，random相关状态

File: paddlenlp/utils/checkpoint.py (hunk 2 above)
Comment (wawltor) [reply paddlenlp/utils/checkpoint.py:1]: 这里的checkpoint主要是checkpoint optimizer和learning_rate，random相关状态

**Pull request closed without merging**